service/src/main/java/uk/gov/hmcts/reform/fpl/service/RemoveOrderService.java

```diff
@@ -62,4 +65,29 @@ public void hideOrder(List<Element<GeneratedOrder>> orders, List<Element<Generat
                 hiddenOrders.add(order);
             });
     }
+
+    public List<Element<Child>> removeFinalOrderPropertiesFromChildren(CaseData caseData) {
+        UUID id = getDynamicListValueCode(caseData.getRemovableOrderList(), mapper);
+
+        Optional<GeneratedOrder> removedOrder = caseData.getOrderFromUUID(id);
+
+        if (removedOrder.isEmpty() || !removedOrder.get().isFinalOrder()) {
```

Comment:
if the removedOrder is empty we have a different issue don't we? The should only get here if there is an order to remove, an exception should probably be thrown otherwise